src/test/librbd/test_mock_fixture.h

```diff
@@ -25,7 +25,11 @@ ACTION_P(CopyInBufferlist, str) {
 }
 
 ACTION_P2(CompleteContext, r, wq) {
+#if defined(__clang__)
+  ContextWQ *context_wq = static_cast<ContextWQ *>(wq);
```

Comment:
@tchaikov 
You might be onto something, last set of lines:
```
# 44 "/usr/include/c++/v1/cstddef" 2 3
# 1 "/usr/include/c++/v1/__nullptr" 1 3
# 18 "/usr/include/c++/v1/__nullptr" 3
# 59 "/usr/include/c++/v1/__nullptr" 3
namespace std
{
    typedef decltype(nullptr) nullptr_t;
}
....
# 2 "test-NULL.cc" 2
nullptr;
```

But I think Clang uses a different C++ lib:
[https://wiki.freebsd.org/NewC++Stack](https://wiki.freebsd.org/NewC++Stack)

Which is were the difference originates ... 
How you/we would like to solve this in Ceph is a question...

I would still vote to use the #if-else-end